### ISSUE TITLE
Logical AND/OR assignment in vue 3 SFC template => [vue/no-parsing-error] Parsing error: Unexpected token =.

### ISSUE THREAD

**Issue by M1CK431** [microsoft/vscode-eslint#1595]
Hi,

While eslint isn't reporting any error using `yarn eslint src/MyComponent.vue`, an error is reported in the editor:
```js
[{
	"resource": "src/MyComponent.vue",
	"owner": "_generated_diagnostic_collection_name_#1",
	"severity": 8,
	"message": "[vue/no-parsing-error]\nParsing error: Unexpected token =.",
	"source": "eslint-plugin-vue",
	"startLineNumber": 48,
	"startColumn": 49,
	"endLineNumber": 48,
	"endColumn": 49
}]
```

Sample `MyComponent.vue` file:
```vue
<template>
  <button @click="tuut &&= false" />
</template>

<script>
export default { data: () => ({ tuut: true }) };
</script>
```

eslint CLI output with the same file:
```sh
yarn eslint src/MyComponent.vue
yarn run v1.22.19
$ /home/user/myProject/node_modules/.bin/eslint src/MyComponent.vue
Done in 1.25s.
```

eslint configuration:
```js
/* eslint-env node */
module.exports = {
  root: true,
  env: { node: true, es2022: true },
  parserOptions: { ecmaVersion: "latest" },
  plugins: ["prettier"],
  extends: [
    "plugin:vue/vue3-strongly-recommended",
    "eslint:recommended",
    "./.eslintrc-auto-import.json",
    "plugin:prettier/recommended"
  ],
  overrides: [
    {
      files: ["*.graphql", "*.gql"],
      parser: "@graphql-eslint/eslint-plugin",
      plugins: ["@graphql-eslint"],
      rules: {
        "@graphql-eslint/no-duplicate-fields": "error"
      }
    }
  ],
  rules: {
    "prettier/prettier": [
      "warn",
      { trailingComma: "none", arrowParens: "avoid" }
    ],
    "no-console": process.env.NODE_ENV === "production" ? "warn" : "off",
    "no-debugger": process.env.NODE_ENV === "production" ? "warn" : "off",
    "no-setter-return": "off",
    "no-empty": ["error", { allowEmptyCatch: true }],

    "vue/multi-word-component-names": "off",
    "vue/no-reserved-component-names": "off",
    "vue/max-attributes-per-line": "off",
    "vue/no-template-shadow": "off",
    "vue/one-component-per-file": "off",
    "vue/require-explicit-emits": "off",
    "vue/singleline-html-element-content-newline": "off",
    "vue/require-default-prop": "off",
    "vue/v-on-event-hyphenation": ["error", "never"],
    "vue/component-tags-order": [
      "warn",
      { order: ["template", "script", "style"] }
    ],
    "vue/no-lone-template": "warn",
    "vue/order-in-components": "warn",
    "vue/this-in-template": "error",
    "vue/block-tag-newline": "warn",
    "vue/component-name-in-template-casing": [
      "warn",
      "PascalCase",
      {
        registeredComponentsOnly: false,
        ignores: ["component", "transition", "fa"]
      }
    ],
    "vue/component-options-name-casing": "warn",
    "vue/custom-event-name-casing": "error",
    "vue/no-empty-component-block": "warn",
    "vue/no-ref-object-destructure": "error",
    // "vue/no-required-prop-with-default": "warn" not released yet 😢
    "vue/no-template-target-blank": "error",
    "vue/no-this-in-before-route-enter": "error",
    "vue/no-unsupported-features": ["error", { version: "^3.2.37" }],
    "vue/no-unused-properties": "warn",
    "vue/no-unused-refs": "warn",
    "vue/no-useless-mustaches": "warn",
    "vue/no-useless-v-bind": "warn",
    "vue/padding-line-between-blocks": "warn",
    "vue/prefer-separate-static-class": "error",
    "vue/prefer-true-attribute-shorthand": "warn",
    "vue/v-for-delimiter-style": "warn"
  }
};
```

Related packages in `package.json`:
```js
{
"devDependencies": {
    "@graphql-eslint/eslint-plugin": "^3.14.0",
    "eslint": "^8.5.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-prettier": "^4.2.1",
    "eslint-plugin-vue": "^9.0.0"
  }
}
```

vscode configuration:
```js
{
    "git.alwaysShowStagedChangesResourceGroup": true,
    "prettier.eslintIntegration": true,
    "gitlens.views.repositories.location": "scm",
    "gitlens.views.fileHistory.location": "scm",
    "gitlens.views.lineHistory.location": "scm",
    "gitlens.views.compare.location": "scm",
    "gitlens.views.search.location": "scm",
    "telemetry.telemetryLevel": "off",
    "editor.tabSize": 2,
    "editor.multiCursorModifier": "ctrlCmd",
    "git.allowForcePush": true,
    "git.branchProtection": [
        "develop"
    ],
    "git.pruneOnFetch": true,
    "github.gitAuthentication": false,
    "html.format.wrapLineLength": 80,
    "vetur.useWorkspaceDependencies": true,
    "workbench.colorTheme": "Atom One Dark",
    "problems.showCurrentInStatus": true,
    "eslint.packageManager": "yarn",
    "javascript.format.semicolons": "remove",
    "typescript.format.semicolons": "remove",
    "typescript.surveys.enabled": false,
    "editor.rulers": [
        80
    ],
    "vetur.ignoreProjectWarning": true,
    "[vue]": {
        "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    },
    "workbench.tree.indent": 14,
    "atomKeymap.promptV3Features": true,
    "sortJSON.contextMenu": {
        "sortJSONReverse": false,
        "sortJSONKeyLength": false,
        "sortJSONKeyLengthReverse": false,
        "sortJSONAlphaNum": false,
        "sortJSONAlphaNumReverse": false,
        "sortJSONValues": false,
        "sortJSONValuesReverse": false,
        "sortJSONType": false,
        "sortJSONTypeReverse": false
    },
    "editor.autoClosingBrackets": "always",
    "editor.formatOnPaste": false,
    "editor.formatOnType": false,
    "editor.formatOnSave": true,
    "editor.formatOnSaveMode": "file",
    "editor.quickSuggestions": {
        "strings": true
    },
    "AllAutocomplete.cycleOpenDocumentsOnLaunch": true,
    "AllAutocomplete.maxItemsInSingleList": 9999,
    "eslint.format.enable": true,
    "eslint.validate": [
        "javascript",
        "graphql",
        "vue"
    ],
    "[javascript]": {
        "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    },
    "[graphql]": {
        "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    },
    "css.format.spaceAroundSelectorSeparator": true,
    "css.lint.zeroUnits": "warning",
    "css.lint.unknownAtRules": "ignore",
    "AllAutocomplete.languageSpecialCharacters": {
        "css": "^[\\.#]",
        "scss": "^[\\.#]",
        "csv": "^[\\: ]"
    },
    "vetur.validation.style": false,
    "vetur.completion.tagCasing": "initial"
}
```

Also, notice that others assignments operators seems working, `&=` for example.
I'm using v2.3.5 (latest as of now).

Can you please fix that?

**Comment by dbaeumer** [comment]
Could you please provide me with a GitHub repository I can clone that demos what you are seeing. This ensures I get the same setup that you are seeing the problem in.

Which version of node do you have installed locally when you run ESLint from the command line.

**Comment by M1CK431** [comment]
> Could you please provide me with a GitHub repository I can clone that demos what you are seeing. This ensures I get the same setup that you are seeing the problem in.
> 
I will try.

> Which version of node do you have installed locally when you run ESLint from the command line.
```sh
$ node --version
v19.5.0
```

**Comment by dbaeumer** [comment]
What you can try also is to use that node version as well when running the extension. To do so use the setting `eslint.runtime` which you can point to a node version to be used.

**Comment by M1CK431** [comment]
Which version should I use? Since that issue was open, I have the same behavior with v19.x, v20.x and v21.x...

**Comment by dbaeumer** [comment]
Can you please provide me with a GitHub repository with a minimal setup I can clone that demos this.

**Comment by M1CK431** [comment]
I'm currently working on an open source project which is not published/released yet but will be later this month. That issue exist in that project too. I will provide the github repository of this project once published so it should be easy for you to clone it and try to use the `&&=` syntaxe in any `<template>` part of any `.vue` file.

**Comment by M1CK431** [comment]
Here it is: https://github.com/M1CK431/url-checker 😃

**Comment by dbaeumer** [comment]
Can you please provide steps as well. And I also need to know which VS Code Vue extension you are using.

**Comment by M1CK431** [comment]
Of course! I'm using the old `Vetur` v0.37.3 (will switch soon on `Volar`).

Steps:
- open any `.vue` file in the `webapp` folder
- in the `<template>` section, add `v-if="(tuut &&= 1)"` on any tag

Example on `src/App.vue` at L.8:
![image](https://github.com/microsoft/vscode-eslint/assets/31622263/f4b327b6-06e7-4d8a-a819-ec960b30237a)

**Comment by dbaeumer** [comment]
This error is not generated by the VS Code ESLint plugin. It seems to come from Vetur. If I switch to Volar everything seems OK for me.

**Comment by M1CK431** [comment]
Ok, good to know, will check when I will switch on Volar. Thanks for the time you spent on this.
